mws/resource_mws_workspaces.go

```diff
@@ -3,354 +3,77 @@ package mws
-// for workspace creation in Accounts API for AWS and GCP. It exits early on AWS and picks only
-// the relevant fields for GCP.
-func (w *Workspace) MarshalJSON() ([]byte, error) {
-	if w.Cloud != "gcp" {
-		return json.Marshal(aWorkspace(*w))
-	}
-	workspaceCreationRequest := map[string]any{
-		"account_id":               w.AccountID,
-		"cloud":                    w.Cloud,
-		"cloud_resource_container": w.CloudResourceBucket,
-		"location":                 w.Location,
-		"workspace_name":           w.WorkspaceName,
-	}
-	if w.NetworkID != "" {
-		workspaceCreationRequest["network_id"] = w.NetworkID
-	}
-	if w.PrivateAccessSettingsID != "" {
-		workspaceCreationRequest["private_access_settings_id"] = w.PrivateAccessSettingsID
-	}
-	if w.GkeConfig != nil {
-		workspaceCreationRequest["gke_config"] = w.GkeConfig
-	}
-	if w.GCPManagedNetworkConfig != nil {
-		workspaceCreationRequest["gcp_managed_network_config"] = w.GCPManagedNetworkConfig
-	}
-	if w.ManagedServicesCustomerManagedKeyID != "" {
-		workspaceCreationRequest["managed_services_customer_managed_key_id"] = w.ManagedServicesCustomerManagedKeyID
-	}
-	if w.StorageCustomerManagedKeyID != "" {
-		workspaceCreationRequest["storage_customer_managed_key_id"] = w.StorageCustomerManagedKeyID
-	}
-	return json.Marshal(workspaceCreationRequest)
+	provisioning.Workspace
+	CustomerManagedKeyID string `json:"customer_managed_key_id,omitempty"` // just for compatibility, will be removed
+	GcpWorkspaceSa       string `json:"gcp_workspace_sa" tf:"computed"`
+	Token                *Token `json:"token,omitempty"`
+	WorkspaceURL         string `json:"workspace_url,omitempty" tf:"computed"`
 }
 
-// Create deploys the workspace and waits till it's properly running.
-// In case of error, it removes the failed deployment and returns the message
-func (a WorkspacesAPI) Create(ws *Workspace, timeout time.Duration) error {
-	if a.client.IsGcp() {
-		ws.Cloud = "gcp"
-	}
-	workspacesAPIPath := fmt.Sprintf("/accounts/%s/workspaces", ws.AccountID)
-	err := a.client.Post(a.context, workspacesAPIPath, ws, &ws)
-	if err != nil {
-		return err
-	}
-	if err = a.WaitForRunning(*ws, timeout); err != nil {
-		log.Printf("[ERROR] Deleting failed workspace: %s", err)
-		if derr := a.Delete(ws.AccountID, fmt.Sprintf("%d", ws.WorkspaceID)); derr != nil {
-			return fmt.Errorf("%s - %s", err, derr)
+// wait for DNS caches to refresh, as sometimes we cannot make
+// API calls to new workspaces immediately after it's created
+func verifyWorkspaceReachable(ctx context.Context, w *databricks.WorkspaceClient) error {
+	return retries.New[struct{}](retries.WithRetryFunc(func(err error) bool {
+		var dnsError *net.DNSError
+		if errors.As(err, &dnsError) {
+			log.Printf("[INFO] workspace is not yet reachable: %s", dnsError)
+			// expected to retry on: dial tcp: lookup XXX: no such host
+			return true
 		}
+		return false
+	})).Wait(ctx, func(ctx context.Context) error {
+		ctx, cancel := context.WithTimeout(ctx, 10*time.Second)
```

Comment:
Might be good to abstract out this timeout on top so it easy to read where the timeouts are defined when navigating code.